nixos/modules/services/audio/mpd.nix

```diff
@@ -152,6 +162,37 @@ in {
         '';
       };
 
+      credentials = mkOption {
+        type = types.listOf (types.submodule {
+          options = {
+            passwordFile = mkOption {
+              type = types.path;
+              description = ''
+                Path to file containing the password.
+              '';
+            };
+            permissions = let
+              perms = ["read" "add" "control" "admin"];
+            in mkOption {
+              type = types.listOf (types.enum perms);
+              default = [ "read" ];
+              description = ''
+                List of permissions that are granted with this password.
+                Permissions can be "${concatStringsSep "\", \"" perms}".
+              '';
+            };
+          };
+        });
+        description = ''
+          Credentials and permissions for accessing the mpd server.
+        '';
+        default = [];
+        example = [
+          {passwordFile = "/var/lib/secrets/mpd_readonly_password"; permissions = [ "read" ];}
+          {passwordFile = "/var/lib/secrets/mpd_admin_password"; permissions = ["read" "add" "control" "admin"];}
+        ];
+      };
+
       credentialsFile = mkOption {
```

Comment:
It would probably make sense to remove this option in favor of the newly introduced option or at least add a deprecation notice.

Reply:
There's this for instance:

https://github.com/NixOS/nixpkgs/blob/57e9cb9112297a0a9ebaf7521f76371ab5971e54/nixos/modules/services/network-filesystems/samba.nix#L64-L72

And:

https://github.com/NixOS/nixpkgs/blob/57e9cb9112297a0a9ebaf7521f76371ab5971e54/lib/modules.nix#L786-L804

Reply:
Since the option has not made it to a stable release yet, maybe we can just completely remove it? Otherwise your suggestion of using `mkRemoveOptionModule` also looks good to me.

Reply:
If an option never made it to stable then no need to deprecate - as you said, just remove. :+1:

Reply:
I've removed it.